_posts/2015-03-23-design-patterns-part-2.markdown

```diff
@@ -0,0 +1,155 @@
+---
+layout:post
+title:'Design Patterns, Fast Tests, and Robust Software, Part II'
+date:2015-02-23 12:01:00
+categories: design patters, testing, software architecture
+---
+
+<meta charset="utf-8">
+<!-- not sure this charset isn't pulled in from _includes --> 
+
+# Part II: Commands, Dependency Injection, Factories
+
+In part I, we talked about using using the Repository and Entity patterns in order to centralize our ORM-specific code.  In this installment, we’ll talk about three other design patterns.  We’ll talk about Command Pattern, which we’ll use to do a lot of the heavy lifting in our app.  And we’ll rely on Dependency Injection to make our life easy when it comes to testing.  Finally, we’ll talk about having our Rails controller interface with a Factory that sets up out context for us.
+
+## Commands and Dependency Injection
+
+Many web apps start small, but the successful ones end up lasting years.  As an app goes through its lifecycle, it accumulates feature after
+feature.  In a Rails app, models that were only 30 lines long quickly grow into 300 lines.  It's suddenly difficult to navigate all that terrain.
+At SitterCity we prefer to write many small commands, or contexts, rather than stuffing all business logic into the model.  These context classes
+will be initialized with any other classes that the context might need.  Within the Rails system, we end up with barebones models that simple manage
+validations and the relationships to other models (has\_many or  belongs\_to code).  And our software consists of a bunch of Plain Old Ruby Objects that capture all the business logic involved in our app.
```

Comment:
For 'plain old ruby object' can we link to something that explains what it is? Also, putting 'PORO' afterwards in parentheses will make it clear later on when you use that shorthand instead.